docs/src/mixed-integer-optimization.jl

```diff
@@ -0,0 +1,83 @@
+
+# # Example: Mixed integer optimization
+#
+# In this example we demonstrate the use of binary, and integer valued
+# variables. We assume that the reader is already familiar with the construction
+# of `ConstraintTree`s; if not, it is advisable to read the previous part of the
+# documentation first.
+
+# The simple problem we will solve is:
```

Comment:
The equation is valid but seems pretty arbitrary, I'd love to have a realistic problem. No worries here, will think of something.

(Ideas: salesman/Hamiltonian? Backpack problem with a few extra constraints?)

Reply:
Ah, actually, I went for super simple to just illustrate it and not have too much supporting code which is tiring to read